File: sliding_window/static_sliding_window.go
package sliding_window

func MaxSum(arraylist []int, window int) int {
	// windowSum holds the total sum of the
	// numbers within the window
	var windowSum int = 0

	// maxSum is the greatest sum within the windows
	maxSum := 0

	for i := 0; i < len(arraylist); i++ {
		// we initialize the window sum with the first element in
		// the array
		windowSum += arraylist[i]

		if windowSum > maxSum {
			maxSum = windowSum
		}

		// removes the first element in the window when the window width is reached
		if i >= window-1 {
			windowSum -= arraylist[i-window+1]
		}
	}
	return maxSum
}

File: sliding_window/static_sliding_window.zig
const std = @import("std");
const print = std.debug.print;

pub fn main() void {
    const arraylist = [5]i32{ 10, -20, -10, 30, 5 };

    const mSum: i32 = maxSum(arraylist, 3);
    print("{} \n", .{mSum});
}

fn maxSum(arraylist: [5]i32, window: usize) i32 {
    var windowSum: i32 = 0;
    var max: i32 = 0;
    var size: usize = 0;

    for (0..arraylist.len) |i| {
        windowSum += arraylist[i];

        if (windowSum > max) {
            max = windowSum;
        }

        if (i >= window - 1) {
            size = window - 1;
            windowSum -= arraylist[i - size];
        }
    }

    return max;
}
